R/clustering.R

```diff
@@ -50,18 +50,22 @@ FindClusters.default <- function(
     clustering.results <- future_lapply(
       X = resolution,
       FUN = function(r) {
-        ids <- RunModularityClustering(
-          SNN = object,
-          modularity = modularity.fxn,
-          resolution = r,
-          algorithm = algorithm,
-          n.start = n.start,
-          n.iter = n.iter,
-          random.seed = random.seed,
-          print.output = verbose,
-          temp.file.location = temp.file.location,
-          edge.file.name = edge.file.name
-        )
+        if(algorithm %in% c(1:3) || tolower(algorithm) == "louvain"){
+          ids <- RunModularityClustering(
+            SNN = object,
+            modularity = modularity.fxn,
+            resolution = r,
+            algorithm = algorithm,
+            n.start = n.start,
+            n.iter = n.iter,
+            random.seed = random.seed,
+            print.output = verbose,
+            temp.file.location = temp.file.location,
+            edge.file.name = edge.file.name
+          )
+        } else if (algorithm== 4 || tolower(algorithm) == "leiden"){
+          ids <- RunLeiden(object)
+        }
```

Comment:
Have an `else` block to throw an error if the user specifies an unrecognized algorithm. This will be a lot nicer than R's default error messages.

Reply:
This has been implemented